apps/aechannel/test/aesc_txs_SUITE.erl

```diff
@@ -2875,6 +2879,90 @@ fp_too_soon(Cfg) ->
+                      channel_pubkey := ChannelPubKey} = Props) ->
+                    Channel = aesc_test_utils:get_channel(ChannelPubKey, S),
+                    Round = aesc_channels:round(Channel),
+                    StateHash = aesc_channels:state_hash(Channel),
+                    Props
+                end,
+                % create contract off-chain
+                create_trees_if_not_present(),
+                set_from(Owner, owner, owner_privkey),
+                create_contract_in_trees(_Round    = ContractCreateRound,
+                                         _Contract = "aens",
+                                         _InitArgs = <<"()">>,
+                                         _Deposit  = 2),
+                % force progress contract on-chain
+                fun(#{contract_id := ContractId} = Props) ->
+                    BinToSign = <<Pubkey/binary, ContractId/binary>>,
+                    <<Word1:256, Word2:256>> = enacl:sign_detached(BinToSign,
+                                                                   Privkey),
+                    Sig = aeu_hex:hexstring_encode(
+                            aeso_data:to_binary({Word1, Word2})),
+                    PreclaimArgs = <<"(", Account/binary, ",",
+                                          CHash/binary, ",",
+                                          Sig/binary,
+                                      ")">>,
+                    (force_call_contract_first(Forcer, <<"preclaim">>,
+                                          PreclaimArgs, FPRound))(Props)
+                end,
+                % ensure all gas is consumed and channel is updated
+                fun(#{state := S,
+                      channel_pubkey := ChannelPubKey,
+                      signed_force_progress := SignedForceProgressTx,
+                      solo_payload := #{update    := Update,
+                                       state_hash := ExpectedStateHash,
+                                       round      := ExpectedRound}} = Props) ->
+                    {_ContractId, Caller} = aesc_offchain_update:extract_call(Update),
+                    TxHashContractPubkey = aesc_utils:tx_hash_to_contract_pubkey(
+                                          aetx_sign:hash(SignedForceProgressTx)),
+                    CallId = aect_call:id(Caller,
+                                          FPRound,
+                                          TxHashContractPubkey),
+                    Call = aect_test_utils:get_call(TxHashContractPubkey, CallId,
+                                                    S),
+                    {_, GasPrice, GasLimit} = aesc_offchain_update:extract_amounts(Update),
+                    %% assert all gas was consumed
+                    GasLimit = aect_call:gas_used(Call),
+                    GasPrice = aect_call:gas_price(Call),
+                    %% the default catch all reason for error
+                    <<>> = aect_call:return_value(Call),
```

Comment:
Would it be possible to assert the return type too - for readability of the test?